tests/typecheck/core/test_storage.yml

```diff
@@ -0,0 +1,39 @@
+- case: test_storage_defaults
+  main: |
+    from django.core.files.storage import default_storage, storages
+
+    reveal_type(default_storage)  # N: Revealed type is "django.core.files.storage.filesystem.FileSystemStorage"
+    reveal_type(storages["default"])  # N: Revealed type is "django.core.files.storage.filesystem.FileSystemStorage"
+    reveal_type(storages["staticfiles"])  # N: Revealed type is "django.contrib.staticfiles.storage.StaticFilesStorage"
+
+- case: test_custom_storages
+  main: |
+    from django.core.files.storage import default_storage, storages
+
+    reveal_type(default_storage)  # N: Revealed type is "myapp.storage.MyDefaultStorage"
+    reveal_type(storages["default"])  # N: Revealed type is "myapp.storage.MyDefaultStorage"
+    reveal_type(storages["custom"])  # N: Revealed type is "myapp.storage.MyStorage"
+    reveal_type(storages["staticfiles"])  # N: Revealed type is "django.contrib.staticfiles.storage.StaticFilesStorage"
+
+  custom_settings: |
+    from django.conf.global_settings import STORAGES as DEFAULT_STORAGES
+
+    STORAGES = {
+        **DEFAULT_STORAGES,
+        "default": {"BACKEND": "myapp.storage.MyDefaultStorage"},
```

Comment:
What will happen if `MyDefaultStorage` is something invalid? Like instance of value `1` or `'abc'`